{{cookiecutter.project_slug}}/clients/web/vue3/src/views/Signup.vue

```diff
@@ -1,3 +1,22 @@
+<script lang="ts" setup>
+import InputField from '@/components/inputs/InputField.vue'
+import LoadingSpinner from '@/components/LoadingSpinner.vue'
+import { useUsers } from '@/composables/use-users'
+import Button from '@/components/Button.vue'
+
+const { register, loading, registerForm: form } = useUsers()
+const onRegister = () => {
+  const createAccountInput = {
+    firstName: form.firstName.value ?? '',
+    lastName: form.lastName.value ?? '',
+    email: form.email.value ?? '',
+    password: form.password.value ?? '',
+    confirmPassword: form.confirmPassword.value ?? '',
+  }
+  register(createAccountInput)
```

Comment:
Shouldn't form validation handle this and we should just be able to pass the form to the API call? Why do we need to manually extract each field?

Reply:
This is a known bad interaction between tn-forms and typescript.
Even after validating the type of the form value is `<type> | undefined`
Ideally, after form validation,  the `form.value` should have the right required field types
I'm [pushing for a good TS update](https://github.com/thinknimble/tn-forms/pull/26) to tn-forms but I don't think it is something that will come soon 😅

Reply:
Yea the issue is that the actual definition of the form field values in the form is 
`<type>|undefined`
Because value is an optional field 
In addition to Dami's work I was thinking we could just add a type caster to avoid the issue